sys/fatfs_diskio/native/native_diskio.c

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (C) 2016 Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup  tests
+ * @{
+ *
+ * @file
+ * @brief    Implementation of FatFs interface that makes use of a fatfs image 
+ *           file instead of hardware to allow FatFs usage/testing on native.
+ *           Based on low level disk I/O module example for FatFs by ChaN, 2016
+ *
+ * @author   Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * @}
+ */
+#define ENABLE_DEBUG    (1)
+#include "debug.h"
+#include "fatfs/diskio.h" /* FatFs lower layer API */
+#include "fatfs/integer.h"
+
+#include <stdio.h>
+#include <time.h>
+#include "periph_conf.h"
+#include "periph/rtc.h"
+#include "xtimer.h"
+#include <stdlib.h>
+#include <errno.h>
+
+/* Complete pending write process (needed at _FS_READONLY == 0) */
+#define CTRL_SYNC           0   
+
+/* Get media size (needed at _USE_MKFS == 1) */
+#define GET_SECTOR_COUNT    1   
+
+/* Get sector size (needed at ) */
+#define GET_SECTOR_SIZE     2   
+
+/* Get erase block size (needed at _USE_MKFS == 1) */
+#define GET_BLOCK_SIZE      3   
+
+/* Inform device that the data on the block of sectors is 
+   no longer used (needed at _USE_TRIM == 1) */
+#define CTRL_TRIM           4   
+
+#define RTC_YEAR_OFFSET   1900

```

Comment:
many of these macros are defined multiple times. can you extract them into a common h file?